### PR TITLE
[SMALLFIX]add an input field in browse page

### PR DESCRIPTION
**Pull request by maobaolong**
add an input field to input the file path in browse page

### CODE REVIEW

**Comment by calvinjia** [comment]
Jenkins, ok to test.

**Comment by calvinjia** [comment]
@maobaolong Thanks for the contribution, for UI related changes, could you please attach a screenshot?

**Comment by maobaolong** [comment]
@calvinjia  This is the screenshot

**Comment by gpang** [comment]
Jenkins, add to whitelist

File: core/server/src/main/webapp/browse-body.jsp [hunk 1]
```diff
@@ -16,6 +16,33 @@
...
+    function changeDir() {
+        var path = getInputPath();
+        path = encodeURI(path);
+        window.location.href="./browse?path="+path;
+    }
+
+    $(document).ready(function () {
+        $("#pathInput").val(currentDir);
+        $("#goBtn").click(changeDir);
+        $("#pathInput").keydown(function (e) {
```
Comment (jsimsa) [inline core/server/src/main/webapp/browse-body.jsp]: document what this part does

File: core/server/src/main/webapp/browse-body.jsp [hunk 2]
```diff
@@ -38,6 +65,10 @@
               <% } %>
             </ul>
```
Comment (waltonryan) [inline core/server/src/main/webapp/browse-body.jsp]: Thanks for the update! I suggest we add the input inside `<div class="navbar-inner">`. This will help consolidate the information and place it where it more makes sense.

This html will do the trick:

```
<div id="pathNav" class="input-append pull-right" style="margin: 5px;margin-right: 45px;">
    <input class="span12" id="pathInput" type="text" placeholder="Navigate to a directory">
    <button class="btn" type="button">Go</button>
</div>
```

screen shot 2017-01-06 at 2 16 31 pm

File: core/server/src/main/webapp/browse-body.jsp [hunk 3]
```diff
@@ -16,6 +16,33 @@
 <%@ page import="static java.net.URLEncoder.encode" %>
 
 <jsp:include page="header-scripts.jsp" />
+
+<script>
```
Comment (waltonryan) [inline core/server/src/main/webapp/browse-body.jsp]: If possible, add this script to the parent jsp page and place above the closing `</body>` tag. If not possible because of the need to set `currentDir` then it's ok.

File: core/server/src/main/webapp/browse-body.jsp [hunk 4]
```diff
@@ -38,6 +65,10 @@
               <% } %>
             </ul>
           </div>
+          <div id="pathNav">
```
Comment (waltonryan) [inline core/server/src/main/webapp/browse-body.jsp]: See comment above about moving `<div id="pathNav">` to inside `<div class="navbar-inner">`

File: core/server/src/main/webapp/browse-body.jsp [hunk 5]
```diff
@@ -16,6 +16,33 @@
...
+    }
+
+    function changeDir() {
+        var path = getInputPath();
+        path = encodeURI(path);
+        window.location.href="./browse?path="+path;
+    }
+
+    $(document).ready(function () {
+        $("#pathInput").val(currentDir);
```
Comment (uronce-cc) [inline core/server/src/main/webapp/browse-body.jsp]: why is this line needed?

File: core/server/src/main/webapp/browse-body.jsp [hunk 6]
```diff
@@ -16,6 +16,33 @@
...
+        $("#goBtn").click(changeDir);
+        $("#pathInput").keydown(function (e) {
+            if(e.keyCode==13){
```
Comment (uronce-cc) [inline core/server/src/main/webapp/browse-body.jsp]: for consistency, use `===` and `!==` for (in)equality check.

File: core/server/src/main/webapp/browse-body.jsp [hunk 7]
```diff
@@ -16,6 +16,33 @@
...
+                changeDir();
+            }
+        });
+        if(base != "./browse") {
```
Comment (uronce-cc) [inline core/server/src/main/webapp/browse-body.jsp]: when will the URL base not be `browse`?

File: core/server/src/main/webapp/browse-body.jsp (hunk 1 above)
Comment (maobaolong) [inline core/server/src/main/webapp/browse-body.jsp]: @jsimsa , Thank you for your reply.
line 34: set input tag's value to variable currentDir's value which id is pathInput. currentDir's value is given by serverside request;
line 35: bind goBtn's click event handle function to changeDir, gotBtn is a button tag;
line 36: bind pathInput's keyDown event handle function, correctly, i want to listen enter key down event.

File: core/server/src/main/webapp/browse-body.jsp (hunk 5 above)
Comment (maobaolong) [inline core/server/src/main/webapp/browse-body.jsp]: I think you mean line 34.
Because of i want to give a current path to the input tag which's id is pathInput.

File: core/server/src/main/webapp/browse-body.jsp (hunk 7 above)
Comment (maobaolong) [inline core/server/src/main/webapp/browse-body.jsp]: The designer of the browse page and browseLog page are the same one, i think.
The http requst of browse page and browseLog page are both use browse.jsp to handle, but as you know, the log page dosen't need a nav bar.
I must thank you for your suggestion.

File: core/server/src/main/webapp/browse-body.jsp (hunk 2 above)
Comment (maobaolong) [inline core/server/src/main/webapp/browse-body.jsp]: Thank your for your modify suggestion. It is look good better than mine. I put it inside the `<div class="navbar-inner">` soon.

File: core/server/src/main/webapp/browse-body.jsp (hunk 4 above)
Comment (maobaolong) [inline core/server/src/main/webapp/browse-body.jsp]: Got it.

File: core/server/src/main/webapp/browse-body.jsp (hunk 7 above)
Comment (maobaolong) [inline core/server/src/main/webapp/browse-body.jsp]: I modify those lines, please see the next commit.

**Comment by maobaolong** [comment]
@uronce-cc, @waltonryan, @jsimsa .
I accept your suggestion. and give a new commit. Please give me a new check. Thank you for your help.

**Comment by waltonryan** [comment]
Thanks @maobaolong, would you mind attaching an updated screenshot?

**Comment by maobaolong** [comment]
@waltonryan ,here is the screenshot.

**Comment by maobaolong** [comment]
Why the test failed?
I just modiy a jsp only. Can anybody tell me why?

**Comment by maobaolong** [comment]
Jenkins, ok to test.

**Comment by maobaolong** [comment]
@apc999, I see you correct the failed unit test. can you help me to invoke a new jenkins test?

**Comment by maobaolong** [comment]
@jsimsa can you help me invoke a new jenkins test,
#4576

**Comment by apc999** [comment]
Jenkins, test this please

**Comment by maobaolong** [comment]
@waltonryan @uronce-cc, i am glad to receive the good news. The failed test has passed. Now you can review my PR again. Thank you for the long time waiting.

**Comment by waltonryan** [comment]
Thanks @maobaolong, this looks good to me. Once one of the engineers approves, they'll be merge it.

**Comment by maobaolong** [comment]
@waltonryan thank you for your LGTM, I will make great improvement in the future.

File: core/server/src/main/webapp/browse-body.jsp [hunk 8]
```diff
@@ -16,6 +16,35 @@
 <%@ page import="static java.net.URLEncoder.encode" %>
 
 <jsp:include page="header-scripts.jsp" />
+
+<script>
+    var currentDir = "<%= request.getAttribute("currentPath").toString()%>";
```
Comment (uronce-cc) [inline core/server/src/main/webapp/browse-body.jsp]: for consistency, put a space before `%>`?

File: core/server/src/main/webapp/browse-body.jsp [hunk 9]
```diff
@@ -16,6 +16,35 @@
...
+    }
+
+    function changeDir() {
+        var path = getInputPath();
+        path = encodeURI(path);
+        window.location.href="./browse?path="+path;
+    }
+
+    $(document).ready(function () {
+        if(base != "./browse") {
```
Comment (uronce-cc) [inline core/server/src/main/webapp/browse-body.jsp]: `!==` and `===` are still not used.

File: core/server/src/main/webapp/browse-body.jsp [hunk 10]
```diff
@@ -37,6 +66,10 @@
                 <li class="active"><a href="./browse?path=<%= encode(request.getAttribute("currentPath").toString(), "UTF-8") %>"><%= escapeHtml(((UIFileInfo) request.getAttribute("currentDirectory")).getName()) %></a></li>
               <% } %>
             </ul>
+            <div id="pathNav" class="input-append pull-right" style="margin: 5px; margin-right: 45px;width:600px;">
```
Comment (uronce-cc) [inline core/server/src/main/webapp/browse-body.jsp]: for consistency, always add a space after `;`.

File: core/server/src/main/webapp/browse-body.jsp (hunk 9 above)
Comment (uronce-cc) [inline core/server/src/main/webapp/browse-body.jsp]: `if (....)`

File: core/server/src/main/webapp/browse-body.jsp [hunk 11]
```diff
@@ -16,6 +16,35 @@
...
+    function changeDir() {
+        var path = getInputPath();
+        path = encodeURI(path);
+        window.location.href="./browse?path="+path;
+    }
+
+    $(document).ready(function () {
+        if(base != "./browse") {
+            $("#pathNav").hide();
+        }else{
```
Comment (uronce-cc) [inline core/server/src/main/webapp/browse-body.jsp]: `} else {`

File: core/server/src/main/webapp/browse-body.jsp [hunk 12]
```diff
@@ -16,6 +16,35 @@
...
+
+    $(document).ready(function () {
+        if(base != "./browse") {
+            $("#pathNav").hide();
+        }else{ 
+            //bind the tag's event handler by given id. 
+            $("#pathInput").val(currentDir);
+            $("#goBtn").click(changeDir);
+            $("#pathInput").keydown(function (e) {
+                if(e.keyCode==13){
```
Comment (uronce-cc) [inline core/server/src/main/webapp/browse-body.jsp]: `===`

File: core/server/src/main/webapp/browse-body.jsp [hunk 13]
```diff
@@ -16,6 +16,35 @@
...
+        var path = getInputPath();
+        path = encodeURI(path);
+        window.location.href="./browse?path="+path;
+    }
+
+    $(document).ready(function () {
+        if(base != "./browse") {
+            $("#pathNav").hide();
+        }else{ 
+            //bind the tag's event handler by given id.
```
Comment (uronce-cc) [inline core/server/src/main/webapp/browse-body.jsp]: `// set currentDir as default value for #pathInput`.

File: core/server/src/main/webapp/browse-body.jsp [hunk 14]
```diff
@@ -16,6 +16,35 @@
...
+        window.location.href="./browse?path="+path;
+    }
+
+    $(document).ready(function () {
+        if(base != "./browse") {
+            $("#pathNav").hide();
+        }else{ 
+            //bind the tag's event handler by given id. 
+            $("#pathInput").val(currentDir);
+            $("#goBtn").click(changeDir);
```
Comment (uronce-cc) [inline core/server/src/main/webapp/browse-body.jsp]: `// when clicking #goBtn or enter the return key,  change directory.`

File: core/server/src/main/webapp/browse-body.jsp (hunk 8 above)
Comment (maobaolong) [inline core/server/src/main/webapp/browse-body.jsp]: done.

File: core/server/src/main/webapp/browse-body.jsp (hunk 9 above)
Comment (maobaolong) [inline core/server/src/main/webapp/browse-body.jsp]: I am so sorry for that I lost those modification. Now I do it.

Comment (maobaolong) [inline core/server/src/main/webapp/browse-body.jsp]: Got it.

File: core/server/src/main/webapp/browse-body.jsp (hunk 11 above)
Comment (maobaolong) [inline core/server/src/main/webapp/browse-body.jsp]: done.

File: core/server/src/main/webapp/browse-body.jsp (hunk 13 above)
Comment (maobaolong) [inline core/server/src/main/webapp/browse-body.jsp]: Thank you for your comment, I copy it before the code line.

File: core/server/src/main/webapp/browse-body.jsp (hunk 14 above)
Comment (maobaolong) [inline core/server/src/main/webapp/browse-body.jsp]: Thank you for your comment, I copy it before the code line.

File: core/server/src/main/webapp/browse-body.jsp (hunk 12 above)
Comment (maobaolong) [inline core/server/src/main/webapp/browse-body.jsp]: done.

File: core/server/src/main/webapp/browse-body.jsp (hunk 10 above)
Comment (maobaolong) [inline core/server/src/main/webapp/browse-body.jsp]: done.

**Comment by maobaolong** [comment]
@uronce-cc, Thank your very much. Your suggestion improve my coding  habits. I give a new commit, please review it again. Thank you again.

**Comment by maobaolong** [comment]
@uronce-cc, thank you for your review before, can you help me to review the new commit again? I have many idea about the webui and starting from here

**Comment by calvinjia** [comment]
@maobaolong Thanks for the contribution!

**Comment by waltonryan** [comment]
Thanks @maobaolong!

**Comment by maobaolong** [comment]
Happy Spring Festival @calvinjia , @waltonryan , @uronce-cc , @jsimsa , @maobaolong
